design/specs/record-page.md

```diff
@@ -1,71 +1,137 @@
 
-The user can access other records from the record page by utilizing the record navigation controls at the top of the page. There is also a 'go to record' option in the record navigation controls, which opens the record selector component.
+- Each widget contains a "View All" link to the right of the title. This link navigates to the Table Page for the widget table pre-filtered to show only the records related to this record. Any ephemeral filter conditions applied to the widget by the user will also be applied within the link to that Table Page.
 
+- The sheet has pagination controls at the bottom. To prevent nested vertical scrolling, the pagination-page-size is a scant `10`! The user can increase it within the pagination UI if needed, but their new setting will not persist. The widget will grow taller to accommodate a larger pagination-page-size, but its `max-height` is `0.9vh` (or similar) to prevent it from being taller than the viewport. This means that nested vertical scrolling will occur if the user sets a large pagination-page-size or has an uncommonly short viewport.
 
-## Related Features
+    The small pagination-page-size admittedly limits the utility of the widget -- but does so for the sake of improving usability of the Record Page as a whole. If the user wants to see the "full" results from the widget, they can click the "View All" link (even choosing to open that link in a new browser tab if they like).
 
-### Record Summary
-Record summaries are strings that represent a record's data. They are specified by users and can include variable values from the record's fields, or include symbols and text characters. 
-[Record Summary Spec](design/specs/record-summary)
\ No newline at end of file
+- The order of widgets within the Record Page is not defined and also not configurable.
+
+### Longer-term goals for widgets, post-alpha
+
+(This section is subject to further design review and task prioritization, but is presented to justify gaps in functionality of the above specs.)
+
+- There are two types of widgets:
+
+    - **Auto-generated widgets** (which are described above, to be implemented first)
+
+    - **Manually generated widgets** (which are more complex and will hopefully be implemented later)
+    
+        Some example use-cases for manually-generated widgets are:
+
+        - Show a patron's currently-checked out books
+        - Show all publications of an author, along with the number of items that the library has for that publication.
+
+- The title is stylized differently for the two types of widgets (perhaps via a lock icon or similar) allowing the user to distinguish them.
+
+- Above all widgets, an "Add Query" button exists to add a new manually-generated widget.
```

Comment:
I'm not sure if we should call this "Query" (maybe "Widget" or "Related Records" or whatever) but we can figure out the name closer to implementation.

Reply:
Yeah I struggled with this too. But that'll be easy to tweak later.